### DISCUSSION TITLE
infrastructure: migrate from prisma 1 cloud to self hosted prisma v3

### DISCUSSION THREAD

**Discussion by crazyoptimist** [prisma/prisma#14184] [Q&A] [answered]
Thanks for your hard work!  

I'm new to Prisma and got into an existing project. What I'm looking for is a way to migrate from prisma 1 cloud, which will be deprecated very soon, to self-hosted prisma v3 infrastructure.  

Code base has been already migrated to use Prisma v3 APIs, so this is specifically for hosting the prisma infrastructure.  
Assuming that we've already hosted our new MySQL database on the AWS RDS, what steps should I follow?   

---

Is this docker image legit for production deployment? `prismagraphql/prisma`

**Comment by ruheni** [comment, marked as answer]
Hi 👋🏽

>Assuming that we’ve already hosted our new MySQL database on the AWS RDS, what steps should I follow?

The next step would be defining the `DATABASE_URL` environment variable for your project in your deployment provider for your API.

If you’re using a relational database in production, you will need to sync your API’s Prisma schema with the database schema before deploying the application using [Prisma Migrate](https://www.prisma.io/docs/concepts/components/prisma-migrate) with `npx prisma migrate deploy` if you have a migration history. `prisma migrate deploy` applies any pending migrations to your database.

You can then generate [Prisma Client](https://www.prisma.io/docs/concepts/components/prisma-client) in the deployment step – which will be used by your APIs – using `npx prisma generate`.

I would recommend taking a look at our [Getting Started guide](https://www.prisma.io/docs/getting-started) to familiarize yourself with Prisma.

> And, Is this docker image legit for production deployment? `prismagraphql/prisma`

The Docker image is old, unmaintained, unsupported, and will be deprecated soon.

Let me know if you have any further questions. 🙂

_Reposted from slack_

**Reply by crazyoptimist** [reply]
Thank you so much for your support in slack! Upgrade is in progress now.  
And it turned out that the docker image above is not needed in the prisma@2 or upper.

**Comment by crazyoptimist** [comment]
Hey! We've migrated the infrastructure in staging env and everything works fine. We now need to migrate the production data into the new database. I've already tried `prisma1 export` and got the dump, which is kinda huge. But trying to import the dump into a new db using `prisma1 import` I'm stuck at `validation error -- unknown field`. I guess it's due to schema evolution since when the upgrade in the code base started.

All in all, I'm looking for a way to connect to the database underlying behind the prisma1 cloud. Then I will be able to dump the db into a plain sql file and import it back to a fresh new db without any hassle.

Do you think it's possible? Or could you suggest another solution for the data migration?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
